File: src/service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Service = void 0;
var data_1 = require("./data");
var Service = /** @class */ (function () {
    function Service() {
        this.data = new data_1.Data();
    }
    Service.prototype.findUserByUsername = function (username) {
        return this.data.findUserData().find(function (user) { return user.getUsername() === username; });
    };
    Service.prototype.findWalletByUserUsername = function (username) {
        return this.data.findWalletData().find(function (wallet) { return wallet.getUser().getUsername() === username; });
    };
    Service.prototype.isAuthenticated = function (username, password) {
        var _a, _b;
        return ((_a = this.findUserByUsername(username)) === null || _a === void 0 ? void 0 : _a.getUsername()) === username
            && ((_b = this.findUserByUsername(username)) === null || _b === void 0 ? void 0 : _b.getPassword()) === password;
    };
    Service.prototype.depositMoney = function (toDeposit, username) {
        var wallet = this.findWalletByUserUsername(username);
        if (!wallet) {
            console.log("Wallet not found for the given username");
            return;
        }
        if (toDeposit < 0) {
            console.log("Deposit must be greater than 0");
        }
        else {
            var updatedMoney = wallet.getMoney() + toDeposit;
            wallet.setMoney(updatedMoney);
            console.log("Deposit of ".concat(toDeposit, " successful. New balance: ").concat(updatedMoney));
        }
    };
    Service.prototype.withDrawMoney = function (toWithdraw, username) {
        var wallet = this.findWalletByUserUsername(username);
        if (!wallet) {
            console.log("Wallet not found for the given username");
            return;
        }
        var actualMoney = wallet.getMoney();
        if (actualMoney < toWithdraw) {
            console.log("You can only retire a value inferior to your money");
        }
        else {
            var updatedMoney = wallet.getMoney() - toWithdraw;
            wallet.setMoney(updatedMoney);
            console.log("Deposit of ".concat(toWithdraw, " successful. New balance: ").concat(updatedMoney));
        }
    };
    return Service;
}());
exports.Service = Service;

File: src/Service.java
import java.util.Optional;

public class Service {
  private final Data data = new Data();

  public void preventChoice(String usernmane) {
    System.out.println("Actual money: " + findWalletByUserUsername(usernmane).getMoney());
    System.out.println("choice: ");
    System.out.println("\n1. deposit");
    System.out.println("\n2. withdraw");
    System.out.println("\n3. exit");
  }

  public User findUserByUsername(String username) {
    Optional<User> foundUser = data.findUserData()
        .stream()
        .filter(user -> user.getUsername().equals(username))
        .findFirst();

    return foundUser.orElse(null);
  }

  public Wallet findWalletByUserUsername(String username) {
    Optional<Wallet> foundWallet = data.findWalletData()
        .stream()
        .filter(wallet -> wallet.getUser().getUsername().equals(username))
        .findFirst();

    return foundWallet.orElse(null);
  }

  public boolean isAuthenticated(String username, String password) {
    User authenticated = findUserByUsername(username);
    return authenticated.getPassword().equals(password);
  }

  public boolean deposit(String username, int toDeposit) {
    if (toDeposit < 0) {
      System.out.println("deposit must be greater than 0");
      return false;
    }
    Wallet wallet = findWalletByUserUsername(username);
    int actualMoney = wallet.getMoney();
    wallet.setMoney(actualMoney + toDeposit);
    System.out.println("transaction completed");
    return true;
  }

  public boolean borrow(String username, int toBorrow) {
    Wallet wallet = findWalletByUserUsername(username);
    int actualMoney = wallet.getMoney();
    wallet.setMoney(actualMoney - toBorrow);
    System.out.println("actual balance: " + wallet.getMoney());
    return true;
  }

  public boolean withdraw(String username, int toWithdraw) {
    Wallet wallet = findWalletByUserUsername(username);
    if (toWithdraw > wallet.getMoney()) {
      System.out.println("to withdraw must be inferior of atcual money");
      return false;
    }
    int actualMoney = wallet.getMoney();
    wallet.setMoney(actualMoney - toWithdraw);
    return true;
  }
}
